### PR TITLE
Fixed JSHint warnings

### PR DESCRIPTION
**Pull request by garrypas**

### CODE REVIEW

**Comment by brianmhunt** [comment]
Thanks @garrypas .

The jshint updates are great, thank you for doing that. I'll try and merge these just after the upcoming 3.3.0 (which should coincide with a gulp build PR #1360 that does auto-jshint).

The native `array.map` is ([by design](http://www.ecma-international.org/ecma-262/5.1/#sec-15.4.4.19)) [much, much slower](http://jsperf.com/map-reduce-named-functions/2) than the `utils.arrayMap` equivalent. Performance aside, is there a benefit you had in mind for this change?

Cheers.

**Comment by garrypas** [comment]
@brianmhunt I just did some benchmarking and you are absolutely correct. That's a big surprise to me. I've therefore reverted the changes related to Array.map().

**Comment by brianmhunt** [comment]
@garrypas Crazy, eh? Thanks for the update.

File: src/subscribables/observableArray.changeTracking.js [hunk 1]
```diff
@@ -107,6 +108,7 @@ ko.extenders['trackArrayChanges'] = function(target) {
 
             case 'pop':
                 offset = arrayLength - 1;
+            break;
```
Comment (mbest) [inline src/subscribables/observableArray.changeTracking.js]: This is wrong (as is the earlier added `break`).

File: src/subscribables/observableArray.changeTracking.js [hunk 2]
```diff
@@ -121,7 +123,7 @@ ko.extenders['trackArrayChanges'] = function(target) {
                     endAddIndex = startIndex + argsLength - 2,
                     endIndex = Math.max(endDeleteIndex, endAddIndex),
                     additions = [], deletions = [];
-                for (var index = startIndex, argsIndex = 2; index < endIndex; ++index, ++argsIndex) {
+                for (index = startIndex, argsIndex = 2; index < endIndex; ++index, ++argsIndex) {
```
Comment (mbest) [inline src/subscribables/observableArray.changeTracking.js]: Where is `index` defined now?

Comment (garrypas) [inline src/subscribables/observableArray.changeTracking.js]: Hi, its on line 104 (was line 103)

Comment (brianmhunt) [inline src/subscribables/observableArray.changeTracking.js]: Linter dividends! :)

**Pull request closed without merging**